transcom/mymove
# INT B-18653 Multi-Moves Routing & Enhancements

INTEGRATION BRANCH

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a904808)

## Summary

With Multiple Moves being integrated and being able to have the functionality of a user being able to not only see their move, but be able to create a new move - we needed to change the current routing in order to support multiple moves.

Currently, the home page of a move was just the URL `/move`. This page used data from state and was not flexible or dynamic with what state held. With the use of the React hook `useParams()`, we can leverage the `moveId` and select the move data from the recently added API endpoint of `getAllMoves` and have the move home page be more dynamic and should allow for multiple home pages for a user.

This PR does the following:
- updated some backend return values (primarily db queries) to support the new move home page
- added `useEffect` to the `MultiMovesLandingPage` that will allow for current data to be loaded and saved into state on component mount
- updated the routing return of `MultiMovesLandingPages` to consider the current profile state 
- updated test data to what is being returned by the `getAllMoves` API endpoint
- implemented early `LoadingPlaceholer` returns as data loads to avoid async errors
- updated `MultiMoveContainer` to show shipment unique identifier instead of move code
- created `MoveHome.jsx` and essentially refactored the previously used `Home` component from a class to a functional component. This will allow for the use of hooks (`useEffect`, `useState`, `useParams`, etc)

**NOTE** the full functionality of adding another move is not yet in place. This will be done with [Multiple Moves - Create Move Button Flow](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a904812). Once that is in place, the user should be able to upload orders AKA moves.

### How to test

1. Access MM as a customer
2. If you do not have a profile set up, you should be routed to the profile creation flow
3. Click `Create a Move` button on the landing page
4. This should navigate you to the basic `/move` home page
5. Click `Add Orders` and add your info
6. Once complete, you should be routed back to the landing page and see your move
7. Click `Go to Move` button and you should see the URL contain the `moveId` params
